feature-libs/checkout/base/components/services/checkout-config.service.ts

```diff
@@ -20,7 +21,30 @@ export class CheckoutConfigService {
   private defaultDeliveryMode: Array<DeliveryModePreferences | string> =
     this.checkoutConfig.checkout?.defaultDeliveryMode || [];
 
-  constructor(private checkoutConfig: CheckoutConfig) {}
+  protected checkoutFlow =
+    this.checkoutFlowOrchestratorService?.getCheckoutFlow();
+
+  // TODO(NO_TICKET): make checkoutFlowOrchestratorService a required dependency
+  constructor(
+    checkoutConfig: CheckoutConfig,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    checkoutFlowOrchestratorService: CheckoutFlowOrchestratorService
+  );
+  /**
+   * @deprecated since 6.6
```

Comment:
What do you think about usage of inject() instead of adding dependency to constructor to avoid breaking changes? You can take a look on Kris last message on tribe for the reference.

Reply:
Good idea, done.